rancher2/resource_rancher2_machine_config_v2.go

```diff
@@ -47,8 +47,21 @@ func resourceRancher2MachineConfigV2Create(d *schema.ResourceData, meta interfac
 		MinTimeout: 3 * time.Second,
 	}
 	_, waitErr := stateConf.WaitForState()
+
+	// Terraform may finish creating the machine config v2 before rancher has set get
+	// permissions on the resource. Retry a few times to see if the state will return
+	// without error.
 	if waitErr != nil {
-		return fmt.Errorf("[ERROR] waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
+		if IsNotAccessibleByID(waitErr) {
+			for i := 0; i < 3; i++ {
+				_, err = stateConf.WaitForState()
+				if err != nil {
+					continue
+				}
+				return flattenMachineConfigV2(d, newObj)
+			}
+		}
+		return fmt.Errorf("[ERROR] Waiting for machine config (%s) to be active: %s", newObj.ID, waitErr)
```

Comment:
Can we just replace `waitErr` with `err`? Sorry should've caught this first pass, just worried about the case where neither `waitErr` nor `err` were nil but for potentially different reasons.

Reply:
I'm going to try and condense the logic so I'll just have an err within the for loop. Pending